src/base/edition.rs

```diff
@@ -703,3 +703,73 @@ unsafe fn extend_rows<N: Scalar>(
         );
     }
 }
+
+/// Extend the number of columns of the `Matrix` with elements from
+/// a given iterator.
+impl<N, R, S> Extend<N> for Matrix<N, R, Dynamic, S>
+where
+    N: Scalar,
+    R: Dim,
+    S: Extend<N>,
+{
+    /// Extend the number of columns of the `Matrix` with elements
+    /// from the given iterator.
+    ///
+    /// # Example
+    /// ```
+    /// use nalgebra::{Dynamic, Matrix, MatrixMN, Matrix3};
+    ///
+    /// let data = vec![0, 1, 2,      // column 1
+    ///                 3, 4, 5];     // column 2
+    ///
+    /// let mut matrix : MatrixMN<_, Dynamic, Dynamic> =
+    ///   Matrix::from_vec_generic(Dynamic::new(3), Dynamic::new(2), data);
```

Comment:
This should be `DMatrix::from_vec(3, 2, data)`.
Perhas this will avoid having to specify the type explicitly above (which should be `DMatrix<_>` instead of `MatrixMN<_, Dynamic, Dynamic>`.